File: apps/React-TS/src/utils/types/risk-score.ts
/**
 * TypeScript types for the risk score and lab values API responses.
 * Mirrors the FastAPI Pydantic schemas in apps/FastAPI/schemas/risk_score.py.
 */

// ── GET /api/v1/risk-score/data ───────────────────────────────────────────────

export interface RiskScoreResponse {
  success: boolean;
  message?: string | null;
  risk_score?: number | null;
  priority_score?: number | null;
  metrics?: Record<string, any> | null;
  independent_risk_variable_scores?: Record<string, any> | null;
  dependent_risk_variable_contributions?: Record<string, any> | null;
}

// ── GET /api/v1/risk-score/lab-values ────────────────────────────────────────

export interface LabValuesResponse {
  hemoglobin_a1c?: number | null;
  ldl?: number | null;
  hdl?: number | null;
  triglycerides?: number | null;
  creatinine?: number | null;
}

File: apps/FastAPI/schemas/risk_score.py
"""
Pydantic schemas for the risk score and lab values features.

Endpoint → schema mapping:
  GET /api/v1/risk-score/data       → RiskScoreResponse
  GET /api/v1/risk-score/lab-values → LabValuesResponse
"""

from __future__ import annotations

from typing import Any, Dict, Optional

from pydantic import BaseModel


# ── GET /data ─────────────────────────────────────────────────────────────────

class RiskScoreResponse(BaseModel):
    """
    Risk score and priority score with supporting metrics.

    Mirrors the JSON returned by:
      POST /healthie/iframe_provider_tab/risk_score/data
    """
    success: bool
    message: Optional[str] = None
    risk_score: Optional[float] = None
    priority_score: Optional[float] = None
    metrics: Optional[Dict[str, Any]] = None
    independent_risk_variable_scores: Optional[Dict[str, Any]] = None
    dependent_risk_variable_contributions: Optional[Dict[str, Any]] = None


# ── GET /lab-values ───────────────────────────────────────────────────────────

class LabValuesResponse(BaseModel):
    """
    Patient lab values extracted from the risk score metrics.lab_data.

    All fields are optional — a value is None when the patient has no
    recorded entry for that lab marker.
    """
    hemoglobin_a1c: Optional[float] = None
    ldl: Optional[float] = None
    hdl: Optional[float] = None
    triglycerides: Optional[float] = None
    creatinine: Optional[float] = None
